src/webots/gui/WbConsole.cpp

```diff
@@ -264,37 +264,50 @@ WbConsole::~WbConsole() {
   gInstance = NULL;
 }
 
-void WbConsole::clear() {
+void WbConsole::clear(bool reset) {
   mEditor->clear();
+  if (reset)
+    resetFormat();
+}
+
+void WbConsole::resetFormat() {
+  mForegroundColor = ansiBlack();
```

Comment:
Why not simply clearing it?
```suggestion
  mForegroundColor.clear();
```